src/main/java/com/google/devtools/build/lib/remote/ByteStreamUploader.java

```diff
@@ -83,10 +84,10 @@
 
   /** Contains the hash codes of already uploaded blobs. **/
   @GuardedBy("lock")
-  private final Set<HashCode> uploadedBlobs = new HashSet<>();
+  private final Set<Digest> uploadedBlobs = new HashSet<>();
```

Comment:
This may have used HashCode because it's more compact. I think you are changing this to get the size as well. The downside is that this could increase memory consumption by 2x.

Reply:
Switched `uploadedBlobs´ back to `HashCode`.